src/devices/Ssd1351/Ssd1351Imaging.cs

```diff
@@ -0,0 +1,139 @@
+        /// <param name="bm">The bitmap to be sent to the display controller note that only Pixel Format Format32bppArgb is supported.</param>
+        public void SendBitmap(Bitmap bm)
+        {
+            SendBitmap(bm, new Point(0, 0), new Rectangle(0, 0, SCREEN_WIDTH_PX, SCREEN_WIDTH_PX));
+        }
+
+        /// <summary>
+        /// Send a bitmap to the ssd1351 display specifying the starting position and destination clipping rectangle.
+        /// </summary>
+        /// <param name="bm">The bitmap to be sent to the display controller note that only Pixel Format Format32bppArgb is supported.</param>
+        /// <param name="updateRect">A rectangle that defines where in the display the bitmap is written. Note that no scaling is done.</param>
+        public void SendBitmap(Bitmap bm, Rectangle updateRect)
+        {
+            SendBitmap(bm, new Point(updateRect.X, updateRect.Y), updateRect);
+        }
+
+        /// <summary>
+        /// Send a bitmap to the ssd1351 display specifying the starting position and destination clipping rectangle.
+        /// </summary>
+        /// <param name="bm">The bitmap to be sent to the display controller note that only Pixel Format Format32bppArgb is supported.</param>
+        /// <param name="sourcePoint">A coordinate point in the source bitmap where copying starts from.</param>
+        /// <param name="destinationRect">A rectangle that defines where in the display the bitmap is written. Note that no scaling is done.</param>
+        public void SendBitmap(Bitmap bm, Point sourcePoint, Rectangle destinationRect)
+        {
+
+            if (bm == null)
+            {
+                throw new ArgumentNullException(nameof(bm));
+            }
+
+            if (sourcePoint == null)
+            {
+                throw new ArgumentNullException(nameof(sourcePoint));
+            }
+
+            if (destinationRect == null)
+            {
+                throw new ArgumentNullException(nameof(destinationRect));
+            }
+
+            if (bm.PixelFormat != PixelFormat.Format32bppArgb)
+            {
+                throw new ArgumentException($"Pixel format {bm.PixelFormat.ToString()} not supported.", nameof(bm.PixelFormat));
+            }
+
+            // get the pixel data and send it to the display
+            SendBitmapPixelData(GetBitmapPixelData(bm, new Rectangle(sourcePoint.X, sourcePoint.Y, destinationRect.Width, destinationRect.Height)), destinationRect);
+
+        }
+
+        /// <summary>
+        /// Convert a bitmap into an array of pixel data suitable for sending to the display
+        /// </summary>
+        /// <param name="bm">The bitmap to be sent to the display controller note that only Pixel Format Format32bppArgb is supported.</param>
+        /// <param name="sourceRect">A rectangle that defines where in the bitmap data is to be converted from.</param>
+        public Span<byte> GetBitmapPixelData(Bitmap bm, Rectangle sourceRect)
```

Comment:
Should this method be public? Seems like people should just call SendBitmap instead right? Or is there a scenario where it would be useful to call this method directly?

Reply:
This is public as it is useful to be able to cache the bitmap data returned. This is used in the sample where a number of bitmaps are read, converted and then displayed sequentially.